### DISCUSSION TITLE
pulsar broker error with Not enough bookies to create ledger

### DISCUSSION THREAD

**Discussion by dxsooo** [milvus-io/milvus#16028] [Q&A and General discussion] [answered]
pulsar broker error with "Not enough bookies to create ledger", and I check bookkeeper status that one of the container error with "Space left on device data/bookkeeper/ledgers/current: 2503421951, Used space fraction: 0.9533473 > threshold 0.95"

It looks like that bookkeeper save too many logs and make its host machine nearly full, and I located the pvc directory and rm some logs. However it still prints error.

I need some help to recover my bookkeeper and the pulsar broker. And it there any configuration to make bookkeeper save fewer logs, as my machine no too much space for my containers

**Comment by xiaofan-luan** [comment, marked as answer]
you should change defaultRetentionTimeInMinutes in pulsar settings so the pulsar will automatically purge data.

If you use helm install we automatically set data purge to 7 days. So far milvus K8s operator didn't set it

**Reply by sloth2012** [reply]
Is it safe to remove ledger logs？
I set the environment variable defaultRetentionTimeInMinutes=4320，and defaultRetentionSizeInMB=8192 （with k8s），but the logs file still keep increasing。
